src/redux/features/profileSlice.ts

```diff
@@ -0,0 +1,108 @@
+// features/profile/profileSlice.ts
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import { RootState } from '../store';
+
+// Define a type for the slice state
+interface ProfileState {
+  profile: any | null; // Consider defining a more specific type for your profile object
+  status: 'idle' | 'loading' | 'succeeded' | 'failed';
+  error: string | null;
+}
+
+// Define the initial state using the `ProfileState` type
+const initialState: ProfileState = {
+  profile: null,
+  status: 'idle',
+  error: null,
+};
+const apiUrl = `${process.env.NEXT_PUBLIC_API_URL}/profile/`; // Adjust the path as needed
+
+export const fetchProfile = createAsyncThunk('profile/fetchProfile', async (_, { getState }) => {
+  const state = getState() as RootState; // Typecasting to RootState to access the entire Redux state
+  const token = state.user.token; // Assuming the token is stored in state.user.token
+
+  const response = await fetch(apiUrl, {
+    method: 'GET',
+    headers: {
+      'Authorization': `Bearer ${token}`, // Use the token from the Redux state
+      'Content-Type': 'application/json'
+    },
+  });
+
+  if (!response.ok) {
+    throw new Error('Failed to fetch profile');
+  }
+  const data = await response.json();
+  console.log('API Response:', data);
+
+  return data;
+});
```

Comment:
As catherine mentioned, it would be better to use the mutation provided by react-redux toolkit.